src/main/java/frc/trigon/robot/misc/objectdetectioncamera/SimulationObjectDetectionCameraIO.java

```diff
@@ -40,127 +30,224 @@ protected SimulationObjectDetectionCameraIO(String hostname) {
 
     @Override
     protected void updateInputs(ObjectDetectionCameraInputsAutoLogged inputs) {
-        final Rotation2d closestObjectYaw = getClosestVisibleObjectYaw(RobotContainer.POSE_ESTIMATOR.getCurrentPose());
+        final Rotation2d closestObjectYaw = getClosestVisibleObjectYaw(RobotContainer.POSE_ESTIMATOR.getCurrentEstimatedPose());
         if (closestObjectYaw == null) {
             inputs.hasTargets = false;
         } else {
             inputs.hasTargets = true;
-            inputs.bestObjectYaw = closestObjectYaw.getDegrees();
-            inputs.visibleObjectsYaw = new double[]{inputs.bestObjectYaw};
+            inputs.visibleObjectsYaw = new Rotation2d[]{closestObjectYaw};
         }
 
-        updateObjectCollection();
-        updateObjectEjection();
-        updateHeldObjectPose();
-        HAS_OBJECTS = heldObject.length != 0;
-        Logger.recordOutput("Poses/GamePieces/HeldGamePiece", heldObject);
-        Logger.recordOutput("Poses/GamePieces/ObjectsOnField", toPosesArray(objectsOnField));
-    }
+        updateHeldGamePiece();
 
-    private void updateHeldObjectPose() {
-        if (heldObject.length == 0)
-            return;
-        heldObject[0] = getHeldObjectPose(RobotContainer.POSE_ESTIMATOR.getCurrentPose());
+        HAS_OBJECTS = heldObject != null;
+        logGamePieces();
     }
 
-    private void updateObjectEjection() {
-        if (heldObject.length == 0 || !isEjecting() || isDelayingEjection)
-            return;
-        isDelayingEjection = true;
-        Commands.getDelayedCommand(0.04, () -> {
-            heldObject = new Pose3d[0];
-            isDelayingEjection = false;
-        }).schedule();
-    }
+    /**
+     * Gets the yaw relative to the robot of the closest visible object.
+     *
+     * @param robotPose the pose of the robot on the field
+     * @return the yaw of the closest visible object
+     */
```

Comment:
explain more about the internals of how this works
this sounds a bit bad
also, not robot, camera